test/testutil.h

```diff
@@ -145,9 +144,14 @@ void set_test_title(const char *title);
 /*
  * Declarations for user defined functions
  */
-void register_tests(void);
-int test_main(int argc, char *argv[]);
-
+int setup_tests(void);
+void cleanup_tests(void);
+
+const char *test_get_program_name(void);
```

Comment:
These should move up, or the comment above will seem to say that these need to be implemented by the user as well.